### DISCUSSION TITLE
Camera event_id

### DISCUSSION THREAD

**Discussion by picotrain77** [tsightler/ring-mqtt#641] [General Discussion]
Hi all

Last few days I have noticed I am not getting and ring camera events via my HA automation, yet when I check ring the events are present and correctly.

I have the following condition that as of a few days ago appeared to work in the sense that it would prevent events being pushed to me if the event ID was the same, I.E. Prevent notification if it was a simple URL update due to expiry etc.

```
condition: template
value_template: >-
  {{ trigger.from_state.attributes.eventId !=
  trigger.to_state.attributes.eventId }}
```

Looking at the traces for one individual camera over the last few days I note that completely new events are all showing the same event_id. Earlier I triggered the camera by moving around and stored the trace and made note of the event_id. 30 Mins later I did the same and again noted that the automation had triggered correctly but not only was the event_id the same in the from.state and the to.state but it was in fact the same event_id for both triggers 30 mins apart.

Am I missing something here?
Has anyone else had similar issues?

Thanks in advance
Pico

**Comment by tsightler** [comment]
Honestly, I have no idea what you are talking about, but this sounds mostly like an HA question and this isn't an HA forum.

Regarding the lack of notification, this is a known ongoing issue with push notifications in ring-client-api that is being tracked here https://github.com/dgreif/ring/issues/1218,  It's quite clear Ring has made some changes but not yet clear what can be done to solve them as the current fixes don't seem to work reliably for everyone.  Investigations continue however.

**Reply by picotrain77** [reply]
> but this sounds mostly like an HA question and this isn't an HA forum.

Sorry, I clicked the link in the HA addon "Ring-MQTT with Video Streaming" that specifically says 
"If you need help with this addon please use the [discussions section](https://github.com/tsightler/ring-mqtt/discussions) on the main [ring-mqtt project](https://github.com/tsightler/ring-mqtt)."

That brings me directly to this page! Maybe someone should update the link if this is not correct.

Where should I discuss these types of things in future.



> Regarding the lack of notification, this is a known ongoing issue

Home assistant is getting the notifications so cant initially see the link myself.

Thanks

**Reply by tsightler** [reply]
Yes, the link is correct, but you are not asking for help with this addon, you are asking for help with a Home Assistant issue.  You basically even say so yourself here:

> Home assistant is getting the notifications so cant initially see the link myself.

If you are sure of this statement, then the addon has already done it's job and passed the state update to HA.  From there, it's all HA, not the addon, and you are generally better off asking HA questions on HA forums.  If you have a problem with the addon, you can ask here, and you might get lucky and someone here knows the answer.

**Reply by picotrain77** [reply]
OK, so where does home assistant get the event_id from?

The event_id as I understand it comes from ring APi and is pushed to MQTT broker by this addon.
So my question regarding getting the same event_id over and over in HA is not a HA problem, I assume its a ring or addon issue which I why I chose to start a discussion and not a issue, simply to see if anyone was aware or is seeing similar issues.

So NO this is not a HA issue as clearly seen by reading my first post.

**Reply by tsightler** [reply]
I have no idea how HA generates an event_id, or where it comes from but it is not generated or sent by ring-mqtt.  Ring-mqtt logs every it sends, do you see any event IDs there?

I believe that ID is generated by HA for any event on the event bus, but again, I'm not an HA expert.

**Reply by picotrain77** [reply]
> but it is not generated or sent by ring-mqtt. Ring-mqtt logs every it sends, do you see any event IDs there?

Yes, I assume what ever is being received by my MQTT broker is being pushed from your addon and therefore the following confirms the event_id as discussed found in my broker:

{
  "recordingUrl": "REDACTED",
  "eventId": "7237540392311193040"
}

This is how ring has always worked as far as my knowledge goes - it uses event ID and each event is given a different event ID.

So definitely not a HA question and or issue here.

EDIT:
Have discovered that snapshots are sent with a 'timestamp' and video is sent with 'eventid'
